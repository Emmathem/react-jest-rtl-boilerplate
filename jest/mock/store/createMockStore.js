import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
const middleware = [thunk]
export default (initialState) => {
  const mockStore = configureMockStore(middleware)(initialState)
  // eslint-disable-next-line no-undef
  mockStore.dispatch = jest.fn(mockStore.dispatch)
  return mockStore
}