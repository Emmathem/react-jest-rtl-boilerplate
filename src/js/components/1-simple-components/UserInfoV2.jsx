import { useSelector } from 'react-redux'

// component-rely-on-reducer
const UserInfoV2 = (props) => {
  const {
    userId = '',
    userName = '',
    userImg = ''
  } = useSelector(state => state.userInfo)

  return (
    <div className="user-info">
      {
        userId && <div className="user-info__user-id">{userId}</div>
      }
      {
        userName && <div className="user-info__user-name">{userName}</div>
      }
      {
        userImg && <div className="user-info__user-img">
          <img src={userImg} />
        </div>
      }
    </div>
  )
}

export default UserInfoV2
