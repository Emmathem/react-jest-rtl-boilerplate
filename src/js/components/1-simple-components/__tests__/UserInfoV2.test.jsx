/* eslint-disable no-undef */
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'  // renderer for snapshot test
import MockProvider from '@jest/mock/MockProvider'
import createMockStore from '@jest/mock/store/createMockStore'
import UserInfoV2 from '../UserInfoV2'

afterEach(cleanup)

describe('UserInfoV2.jsx', () => {
  it('snapshot renders correctly, truthy values', () => {
    const store = createMockStore({
      userInfo: {
        userId: '202200001',
        userName: 'Test User',
        userImg: './test_user.img'
      }
    })
    const tree = renderer
      .create(<MockProvider mockStore={store}>
        <UserInfoV2 />
      </MockProvider>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('snapshot renders correctly, falsy values', () => {
    const store = createMockStore({
      userInfo: {
        userId: '',
        userName: '',
        userImg: ''
      }
    })
    const tree = renderer
      .create(<MockProvider mockStore={store}>
        <UserInfoV2 />
      </MockProvider>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
