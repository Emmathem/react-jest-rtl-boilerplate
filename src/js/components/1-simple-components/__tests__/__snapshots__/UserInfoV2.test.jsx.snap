// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoV2.jsx snapshot renders correctly, falsy values 1`] = `
<div
  className="user-info"
>
  
  
  
</div>
`;

exports[`UserInfoV2.jsx snapshot renders correctly, truthy values 1`] = `
<div
  className="user-info"
>
  <div
    className="user-info__user-id"
  >
    202200001
  </div>
  <div
    className="user-info__user-name"
  >
    Test User
  </div>
  <div
    className="user-info__user-img"
  >
    <img
      src="./test_user.img"
    />
  </div>
</div>
`;
